src/test/ui/lifetimes/issue-105227.rs

```diff
@@ -0,0 +1,22 @@
+// Regression test for issue #105227.
+
+// //run-rustfix
+
+fn chars0(v :(& str, &str)) -> impl Iterator<Item = char> {
+//~^ HELP to declare that `impl Iterator<Item = char>` captures `'_`, you can consider introducing a named lifetime parameter `'a`
+//~| HELP
+    v.0.chars().chain(v.1.chars())
+    //~^ ERROR hidden type for `impl Iterator<Item = char>` captures lifetime that does not appear in bounds
+    //~| ERROR
+}
+
+fn chars1(v0 : & str, v1 : &str) -> impl Iterator<Item = char> {
```

Comment:
Can you add another test where generics are already in place, something like

```
fn chars2<'a>(v0: &str, v1: &str, x: &'a str) -> (impl Iterator<Item = char>, &'a str) {
```

Reply:
I've added the test case like this:
`fn chars2<'b>(v0: &str, v1: &'_ str, x: &'b str) -> (impl Iterator<Item = char>, &'b str) {`
to test the existing named lifetime parameter, and elided with `'_` case `v1: &'_ str`.